### ISSUE TITLE
False positive: post-fix operator definition seen as a C-style cast

### ISSUE THREAD

**Issue by Farwaykorse** [google/styleguide#481]
Consider the following code in `Source.cpp`, implementing a post-fix `operator++` in terms of the pre-fix operator.
````c++
// Copyright [year] <Copyright Owner>
class example {
  int value{};

 public:
  constexpr example& operator++() noexcept;
  constexpr example operator++(int) noexcept;
};

constexpr example& example::operator++() noexcept {
  ++value;
  return *this;
}

constexpr example example::operator++(int) noexcept {  // <-- line 15
  const example pre{*this};
  operator++();
  return pre;
}
````
Running cpplint:
``````
$ cpplint Source.cpp
Source.cpp:15:  Using C-style cast.  Use static_cast<int>(...) instead  [readability/casting] [4]
Done processing Source.cpp
Total errors found: 1
``````
The same goes for `operator--(int)`.

**Comment by vapier** [comment]
public cpplint development in this repository has been halted (see https://github.com/google/styleguide/pull/837). please migrate to https://github.com/cpplint/cpplint now.
